examples/compare_connectivity_over_time_over_trial.py

```diff
@@ -0,0 +1,392 @@
+                              spectral_connectivity_time)
+from mne.datasets import sample
+
+print(__doc__)
+
+###############################################################################
+# Background
+# ----------
+#
+# Spectral connectivity is a method for inferring the relationship between
+# channels decomposed at different frequencies of interest. The channels
+# could be M/EEG sensors or brain regions estimated with source localization.
+#
+# There are multiple different spectral connectivity measures, e.g. coherence,
+# imaginary part of coherence, phase locking value, and envelope correlations.
+# Additionally, there are also multiple methods to estimate the frequency
+# content, e.g. Fourier transforms, Morlet wavelets and multitapers.
+#
+# In this example, we focus on the two functions
+# :func:`mne_connectivity.spectral_connectivity_epochs` and
+# :func:`mne_connectivity.spectral_connectivity_time`,
+# which both computes connectivity from epoched data :class:`mne.Epochs`.
+#
+# Both functions contain the options to choose the connectivity measurements
+# of interest and the method to decompose the frequency content.
+# The crucial difference between when to use which function lies on
+# the experimental setup and type of tasks.
+#
+# If the data is obtained for repeated trials of a time-locked event,
+# e.g. ERP data, then :func:`mne_connectivity.spectral_connectivity_epochs`
+# is most likely the corresponding function to utilize.
+#
+# If the data is obtained from resting-state or free-moving tasks,
+# e.g. a mirror-game paradigm, then
+# :func:`mne_connectivity.spectral_connectivity_time` might be better suited.
+#
+# The way connectivity is computed for the two functions are slightly different,
+# thus their interpretations and the hypotheses being tested are also different.
+#
+# Connectivity over trials, as computed by
+# :func:`mne_connectivity.spectral_connectivity_epochs` assume epochs have
+# been created around each trial and the function estimates the connectivity
+# at each timepoint over all the repeated trials for the same event.
+# This gives a high temporal resolution, which is often desired for ERP
+# analysis where you are interested in when an effect occurs.
+# However, this approach is not feasible for single-trial data and will
+# result in errorful values when computed on one epoch.
+#
+# On the other hand, if you are interested in determining connectivity over
+# time for single-trials or from experiments that do not involve exactly
+# repeated trials, then :func:`mne_connectivity.spectral_connectivity_time`
+# should be employed. This function also takes data in the form of
+# :class:`mne.Epochs`, but it may consist of a single epoch.
+# If there are multiple epochs, the connectivity over time is computed for
+# epoch separately, with the option to average over epochs.
+#
+# To better illustrate the differences between the two functions,
+# we will employ them on two simulated cases.
```

Comment:
Wdyt about adding this to the top section above the imports? E.g. right after

"A brief background on the difference between the two conditions is provided,
 followed by examples on simulated data and real EEG data."

Reply:
Sure, I'll move it to the top